### ISSUE TITLE
RestOauthController fails to encode errors because of issue with DefaultCallbackErrorHandler

### ISSUE THREAD

**Issue by snimavat** [alvarosanchez/grails-spring-security-rest#494]
[RestOauthController](https://github.com/grails-plugins/grails-spring-security-rest/blob/5a78c55fad51334cee2f29ba0895283840994c7a/spring-security-rest/grails-app/controllers/grails/plugin/springsecurity/rest/RestOauthController.groovy#L103) Would fail with below error when it encounters UserNotFoundException

` java.lang.Integer.encodeAsURL() is applicable `

Because [Default call back error handler](https://github.com/grails-plugins/grails-spring-security-rest/blob/5a78c55fad51334cee2f29ba0895283840994c7a/spring-security-rest/src/main/groovy/grails/plugin/springsecurity/rest/error/DefaultCallbackErrorHandler.groovy#L39) puts integer value in params as `  params.error = FORBIDDEN.value()` and we cant do intValue.encodeAsUrl

In short, if DefaultCallbackErrorHandler.groovy puts any integer values in returned map, it would cause error at this line [this line ](https://github.com/grails-plugins/grails-spring-security-rest/blob/5a78c55fad51334cee2f29ba0895283840994c7a/spring-security-rest/grails-app/controllers/grails/plugin/springsecurity/rest/RestOauthController.groovy#L103)

We have Custom Oauth user details service which throws UserNotFoundException if the oauth user does not exist in our database.

**Comment by snimavat** [comment]
Looks like codecs broken, encodeAsUrl fails on any object. None of the codecs works.
No idea why, 

I had to put @Import([CodecsConfiguration])  on Application.groovy to get codecs working.
All good after that.

**Comment by jdaugherty** [comment]
in #539  & #540 I've gone ahead and fixed this to explicitly refer to the encoder so in the future this won't happen.
